.github/workflows/e2e-tests.yml

```diff
@@ -287,84 +287,116 @@ jobs:
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-and-phpstan-version-config
               composer install --ignore-platform-reqs
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-max-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-max-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-open-end-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-version-v5
               composer install --ignore-platform-reqs
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-version-v7
               composer install --ignore-platform-reqs
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-min-version
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-no-versions
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-version-config-invalid
               OUTPUT=$(../bashunit -a exit_code "1" ../../bin/phpstan)
               echo "$OUTPUT"
               ../bashunit -a contains 'Invalid configuration' "$OUTPUT"
               ../bashunit -a contains 'Invalid PHP version range: phpVersion.max should be greater or equal to phpVersion.min.' "$OUTPUT"
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-version-config-patch
               composer install --ignore-platform-reqs
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
           - script: |
               cd e2e/composer-version-config
               composer install
               ../../bin/phpstan analyze test.php --level=0
+            os: "ubuntu-latest"
+          - script: |
+              cd e2e/bug-12972
+              ../../bin/phpstan analyze
+            os: "macos-latest"
 
     steps:
       - name: "Checkout"
         uses: actions/checkout@v4
 
+      - name: "Install GNU Patch on macOS" # see https://github.com/cweagans/composer-patches/issues/326
+        if: runner.os == 'macOS'
+        run: |
+          brew install gpatch
+          echo "/opt/homebrew/opt/gpatch/libexec/gnubin" >> $GITHUB_PATH
+
```

Comment:
on macOS we need a GNU compatible `patch` command, the the apple-shipped `patch` command cannot apply our patch files:

```
Error: Cannot apply patch 0 (patches/cloudflare-ca.patch)!

In Patches.php line 331:
                                                       
  Cannot apply patch 0 (patches/cloudflare-ca.patch)!  
```